File: Python/Shortest_Palindrome.py
class Solution:
    def shortestPalindrome(self, s: str) -> str:
        if not s:
            return " "

        temp=s+"#"+s[::-1]
        l=[0]*len(temp)
        j=0
        for i in range(1,len(temp)):
            while j>0 and temp[i]!= temp[j]:
                j=l[j-1]

            if temp[i]==temp[j]:
                j+=1
            l[i]=j
            print(l, j, i)
        return s[l[-1]:][::-1]+s

      

File: TypeScript/Shortest Palindrome.ts
class Solution {
    shortestPalindrome(s: string): string {
        function shortestPalindrome(s: string): string {
            if (!s) {
                   return " ";
               }
       
               const temp = s + "#" + s.split("").reverse().join("");
               const lps: number[] = new Array(temp.length).fill(0);
               let j = 0;
       
               for (let i = 1; i < temp.length; i++) {
                   while (j > 0 && temp[i] !== temp[j]) {
                       j = lps[j - 1];
                   }
                   if (temp[i] === temp[j]) {
                       j++;
                   }
                   lps[i] = j;
               }
       
               return s.slice(lps[lps.length - 1]).split("").reverse().join("") + s;
       };
    }
}